commcare_connect/opportunity/tasks.py

```diff
@@ -444,3 +448,96 @@ def fetch_exchange_rates(date=None, currency=None):
         # Parsing it to decimal otherwise the returned object rate will still be in float.
         rate = Decimal(rates[currency])
         return ExchangeRate.objects.create(currency_code=currency, rate=rate, rate_date=date)
+
+
+@celery_app.task()
+def issue_user_credentials():
```

Comment:
It's a bit hard to understand the performance implications of this task.

It seems to create `UserCredential` objects, but in that, for each (CredentialConfiguration, Opportunity), it's firing queries via get_delivery_user_credentials (and the learn one) on `CompletedWork` and `UserCredential` objects.

We should probably take another approach 
1. Either some sort of change log process that listens for changes and creates necessary objects.
2. Or do whatever you are doing a in bulk. i.e. figure out what to create in minimal calls, instead of one by one for each CredentialConfiguration.

Or May be there are only very few `CredentialConfiguration` objects, and we probably don't need to worry about performance?